channeldb/migration12/invoices.go

```diff
@@ -36,7 +36,22 @@ const (
 	acceptTimeType   tlv.Type = 9
 	resolveTimeType  tlv.Type = 11
 	expiryHeightType tlv.Type = 13
-	stateType        tlv.Type = 15
+	htlcStateType    tlv.Type = 15
+
+	memoType        tlv.Type = 0
+	payReqType      tlv.Type = 1
+	createTimeType  tlv.Type = 2
+	settleTimeType  tlv.Type = 3
+	addIndexType    tlv.Type = 4
+	settleIndexType tlv.Type = 5
+	preimageType    tlv.Type = 6
+	valueType       tlv.Type = 7
+	cltvDeltaType   tlv.Type = 8
+	expiryType      tlv.Type = 9
+	paymentAddrType tlv.Type = 10
+	featuresType    tlv.Type = 11
+	invStateType    tlv.Type = 12
+	amtPaidType     tlv.Type = 13
```

Comment:
Can we use this migration to also add the payment hash as a proper field here? Especially for hodl invoices, there is some awkwardness to get to that value.

Reply:
Idea came up in conversation with @bhandras

Reply:
the payment hash is already serialized in the invoice index, so preferably we would just extract it from there instead of serializing again. unfortunately we only track `hash => invoice number`, so it's not so simple to recover the hash from the invoice on it's own.

> Especially for hodl invoices, there is some awkwardness to get to that value.

How is it extracted today?

Reply:
In `CreateRPCInvoice`, the hash is decoded again from the payment request. But maybe we should only add `PaymentHash` to the in-memory invoice and make sure we always populate it in `LookupInvoice`, `FetchAllInvocies`, `QueryInvoices`, etc?

Reply:
yeah i think that'd better than serializing it again on the invoice, though it may be simpler to do that

Reply:
i think that can be done without a migration, so it should be okay to defer for now